rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -108,34 +108,34 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
     client_handles_.clear();
   }
 
-  void revalidate_handles()
+  void remove_null_handles()
   {
-    {
-      VectorRebind<void *> temp;
-      for (auto & subscriber_handle : subscriber_handles_) {
-        if (subscriber_handle) {
-          temp.push_back(subscriber_handle);
-        }
-      }
-      subscriber_handles_.swap(temp);
+    auto it = std::remove(
+      subscriber_handles_.begin(),
+      subscriber_handles_.end(),
+      nullptr
+      );
+    if (it != subscriber_handles_.end()) {
+      subscriber_handles_.erase(it, subscriber_handles_.end());
```

Comment:
Since it is save to call `erase(end(), end())` I would suggest to drop the condition. For readability I think running `erase` and `remove` in a single line (as in the documented examples http://en.cppreference.com/w/cpp/algorithm/remove#Examples) would improve readability even more:

    subscriber_handles_.erase(
        std::remove(subscriber_handles_.begin(), subscriber_handles_.end(), nullptr),
        subscriber_handles_.end());